src/AzureIoTHub.Portal/Server/Controllers/v1.0/DevicesControllerBase.cs

```diff
@@ -114,7 +98,7 @@ public abstract class DevicesControllerBase<TListItem, TModel> : ControllerBase
             {
                 Items = result.Items.Select(x => this.deviceTwinMapper.CreateDeviceListItem(x)),
                 TotalItems = result.TotalItems,
-                NextPage = nextPage
+                NextPage = result.NextPage
```

Comment:
The ``result.NextPage``  returns the internal URL from the registry not the public Url from the portal. Are you sure ?